sympy/sets/sets.py

```diff
@@ -967,11 +969,10 @@ def as_relational(self, x):
         return And(left, right)
 
     def _eval_Eq(self, other):
-        if not other.is_Interval:
-            if (other.is_Union or other.is_Complement or
-                other.is_Intersection or other.is_ProductSet):
-                return
-
+        if not isinstance(other, Interval):
+            if isinstance(other, Set):
```

Comment:
The only thing this (apparently) returns is None, so are the inner if-s necessary? or was there an error in one of the return values?

Reply:
Good catch. I originally replaced the Union/Complement/Intersection/Product -> None with just Set, but I later had to add FiniteSet as an exception and I guess I messed up the logic. I pushed a commit that fixed it.